pallets/exchange/src/tests.rs

```diff
@@ -707,22 +707,22 @@ fn discount_tests_with_discount() {
 		<Exchange as OnFinalize<u64>>::on_finalize(9);
 
 		// Check final account balances
-		assert_eq!(Currency::free_balance(asset_a, &user_2), 1_000_000_000_000_000);
-		assert_eq!(Currency::free_balance(asset_b, &user_2), 1000000500000000);
+		assert_eq!(Currency::free_balance(asset_a, &user_2), 1000000000000000);
+		assert_eq!(Currency::free_balance(asset_b, &user_2), 1000000000000000);
 
-		assert_eq!(Currency::free_balance(asset_a, &user_3), 1000000149700000);
-		assert_eq!(Currency::free_balance(asset_b, &user_3), 799999700000000);
+		assert_eq!(Currency::free_balance(asset_a, &user_3), 1001480000000000);
+		assert_eq!(Currency::free_balance(asset_b, &user_3), 997000000000000);
 
-		assert_eq!(Currency::free_balance(asset_a, &user_4), 700000000000000);
-		assert_eq!(Currency::free_balance(asset_b, &user_4), 733271262753542);
+		assert_eq!(Currency::free_balance(asset_a, &user_4), 998500000000000);
+		assert_eq!(Currency::free_balance(asset_b, &user_4), 1002994000000000);
 
 		// Check final pool balances
-		assert_eq!(Currency::free_balance(asset_a, &pair_account), 300000413062901);
-		assert_eq!(Currency::free_balance(asset_b, &pair_account), 266728537246458);
+		assert_eq!(Currency::free_balance(asset_a, &pair_account), 100020000000000);
+		assert_eq!(Currency::free_balance(asset_b, &pair_account), 200006000000000);
 
-		assert_eq!(Currency::free_balance(HDX, &user_1), 999860000210000);
-		assert_eq!(Currency::free_balance(HDX, &user_2), 599999999300000);
-		assert_eq!(Currency::free_balance(HDX, &user_3), 600000000000000);
+		assert_eq!(Currency::free_balance(HDX, &user_1), 1000000000000000);
```

Comment:
this looks like discount doesn't work?

Reply:
Wait.  I thought  that algorithm has been fixed and verified on few other testcases. And what needed to be done is just to adjust the rest of the tests

So this would mean there are still issues ? 

Ok, I will g through the test cases and validate each one ( to see if it appears to be correct  )

Reply:
I did not test discount per se. I tested only basic 1v1 trade and made sure it's working. There are still comments in order matching with some potential improvements.

Reply:
Multiple trades also not being tested for correctness.

Reply:
test updated. the values looks ok to me now

Reply:
I will do new test cases with multple trades when i will be adding event checks as part of an issue.